### ISSUE TITLE
Custom directory layout (initial support)

### ISSUE THREAD

**Issue by borekb** [versionpress/versionpress#565]
For instance, [Bedrock](https://roots.io/bedrock/) (the roots.io project) uses a folder structure like this:

```
├── composer.json
├── config
│   ├── application.php
│   └── environments
│       ├── development.php
│       ├── staging.php
│       └── production.php
├── vendor
└── web
    ├── app
    │   ├── mu-plugins
    │   ├── plugins
    │   ├── themes
    │   └── uploads
    ├── wp-config.php
    ├── index.php
    └── wp
```

> Note: Bedrock also uses custom configuration system so we cannot support it as a whole yet, however, the **structure** is something we possibly could.

We've also had similar request from e-labo.biz who use a similar structure but without the Composer files and with `index.php` and `wp-config.php` inside the `app` folder.
- _This issue is blocked by #630_

**Comment by LukeAbell** [comment]
Would love to get bedrock support for versionpress!

**Comment by borekb** [comment]
@LukeAbell Bedrock's structure is supported in v3.0 but their configuration system is yet another challenge and does not work with VP at the moment. With that being said, I am personally a big fan of everything that the Roots.io guys do and it would certainly be nice to make the two projects work well together one day.

**Comment by LukeAbell** [comment]
Any movement here?

**Comment by borekb** [comment]
This issues is done at the scope described in https://github.com/versionpress/versionpress/pull/694#issue-131021836. If you have new specific requirements, feel free to open a new issue.

**Comment by swalkinshaw** [comment]
@borekb can you give any more details on the issues with VP and Bedrock's "configuration system"?

We just load a couple additional files but `wp-config.php` still exists as needed.

**Comment by borekb** [comment]
I'd need to refresh my memory but I think it was that VersionPress has [its own approach to multi-env configuration](https://docs.versionpress.net/en/getting-started/configuration) which is not entirely compatible with Bedrock. For example, VP needs to carefully gitignore what needs to be local (DB connection, for example) vs. what is shared.

**Comment by swalkinshaw** [comment]
Interesting...

So here's what I can gather:

* `wp-config.php` is ignored because it might contain sensitive data
* VP injects and uses `wp-config.common.php` to parse out some constants

In Bedrock, `wp-config.php` just acts as a loader so there's nothing sensitive in it anyway but it is version controlled. I don't know if that's a conflict.

Bedrock requires `config/application.php` which is basically equivalent to `wp-config.common.php` and can contain the constants VP needs.

Interestingly, `WpConfigSplitter` can accept a custom config path. Theoretically it could take `config/application.php` and everyone can be happy. But there's the issue of the include statement for  `wp-config.common.php` which is hardcoded and would conflict.

Bedrock's is obviously different: https://github.com/roots/bedrock/blob/f7f3655829e7f9339b965585539e2570103223ef/web/wp-config.php#L8

**Comment by borekb** [comment]
Great analysis, I think the two will be made work together at some point, there shouldn't be any major roadblocks.

**Comment by acki** [comment]
+1

**Comment by juslintek** [comment]
Tell me what is stopping from making this work, maybe I can fork or you can make a branch for me. I really need this feature.

**Comment by borekb** [comment]
@juslintek Same as https://github.com/versionpress/versionpress/issues/1029#issuecomment-379291624 :)
